src/vs/editor/common/editorContextKeys.ts

```diff
@@ -28,6 +28,7 @@ export namespace EditorContextKeys {
 
 	// -- mode context keys
 	export const languageId = new RawContextKey<string>('editorLangId', undefined);
+	export const fileExtension = new RawContextKey<string>('editorFileExtension', undefined);
```

Comment:
Don't use `File` as it might not a file, maybe `editorResourceExtension` or `editorExtension`?